gotham/src/router/builder/mod.rs

```diff
@@ -267,6 +267,382 @@ where
+    P: Send + Sync + 'static,
+    PE: PathExtractor + Send + Sync + 'static,
+{
+    node_builder: &'a mut NodeBuilder,
+    pipeline_chain: C,
+    pipelines: PipelineSet<P>,
+    phantom: PhantomData<PE>,
+}
+
+impl<'a, C, P, PE> AssociatedRouteBuilder<'a, C, P, PE>
+where
+    C: PipelineHandleChain<P> + Copy + Send + Sync + 'static,
+    P: Send + Sync + 'static,
+    PE: PathExtractor + Send + Sync + 'static,
+{
+    /// Binds a new `PathExtractor` to the associated routes.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # #[macro_use]
+    /// # extern crate log;
+    /// # extern crate gotham;
+    /// # #[macro_use]
+    /// # extern crate gotham_derive;
+    /// # extern crate hyper;
+    /// #
+    /// # use hyper::Response;
+    /// # use gotham::router::Router;
+    /// # use gotham::router::builder::*;
+    /// # use gotham::state::State;
+    /// #
+    /// fn handler(_state: State) -> (State, Response) {
+    ///     // Implementation elided.
+    /// #   unimplemented!()
+    /// }
+    ///
+    /// #[derive(StateData, PathExtractor, StaticResponseExtender)]
+    /// struct MyPathExtractor {
+    /// #   #[allow(dead_code)]
+    ///     id: u32,
+    /// }
+    ///
+    /// #
+    /// # fn router() -> Router {
+    /// build_simple_router(|route| {
+    ///     route.associate("/resource/:id", |assoc| {
+    ///         let mut assoc = assoc.with_path_extractor::<MyPathExtractor>();
+    ///         assoc.get().to(handler);
+    ///     });
+    /// })
+    /// # }
+    /// # fn main() { router(); }
+    /// ```
+    pub fn with_path_extractor<'b, NPE>(&'b mut self) -> AssociatedRouteBuilder<'b, C, P, NPE>
+    where
+        NPE: PathExtractor + Send + Sync + 'static,
+    {
+        AssociatedRouteBuilder {
+            node_builder: self.node_builder,
+            pipeline_chain: self.pipeline_chain,
+            pipelines: self.pipelines.clone(),
+            phantom: PhantomData,
+        }
+    }
+
+    /// Associates a route which matches requests with any of the specified methods, to the current
+    /// path.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate gotham;
+    /// # extern crate hyper;
+    /// #
+    /// # use hyper::{Response, Method};
+    /// # use gotham::router::Router;
+    /// # use gotham::router::builder::*;
+    /// # use gotham::state::State;
+    /// #
+    /// fn handler(_state: State) -> (State, Response) {
+    ///     // Implementation elided.
+    /// #   unimplemented!()
+    /// }
+    ///
+    /// #
+    /// # fn router() -> Router {
+    /// build_simple_router(|route| {
+    ///     route.associate("/resource", |assoc| {
+    ///         assoc.request(vec![Method::Get, Method::Head, Method::Post]).to(handler);
+    ///     });
+    /// })
+    /// # }
+    /// # fn main() { router(); }
+    /// ```
+    pub fn request<'b>(&'b mut self, methods: Vec<Method>) -> DefaultSingleRouteBuilder<'b, C, P> {
```

Comment:
Self-review: Might be useful if the implementation actually **used** the `PE` type parameter.

😳

Reply:
:beer: